source/Mlos.Python/mlos/OptimizerEvaluationTools/unit_tests/TestOptimizerEvaluator.py

```diff
@@ -0,0 +1,218 @@
+        num_pickled_optimizers = len([name for name in os.listdir(pickled_optimizers_dir) if os.path.isfile(os.path.join(pickled_optimizers_dir, name))])
+        self.assertTrue(num_pickled_optimizers == 11)
+
+        # Finally, let's make sure that the optimizers serialized to disk are usable.
+        #
+        with open(os.path.join(pickled_optimizers_dir, "99.pickle"), "rb") as in_file:
+            final_optimizer_from_disk = pickle.load(in_file)
+        final_optimizer_from_report = pickle.loads(optimizer_evaluation_report.pickled_optimizers_over_time[99])
+
+        for _ in range(100):
+            self.assertTrue(final_optimizer_from_disk.suggest() == final_optimizer_from_report.suggest())
+
+
+    def test_named_configs(self):
+        """Tests named optimizer configurations against named objective functions.
+
+        It is prohibitively expensive to test the entire cross product so we test only its subset, but in such a way that
+        each configuration will be tested at least once.
+        """
+        optimizer_named_configs = bayesian_optimizer_config_store.list_named_configs()
+        num_optimizer_configs = len(optimizer_named_configs)
+        objective_function_named_configs = objective_function_config_store.list_named_configs()
+        num_objective_function_configs = len(objective_function_named_configs)
+
+        num_tests = max(num_optimizer_configs, num_objective_function_configs)
+
+        with traced(scope_name="parallel_tests"), concurrent.futures.ProcessPoolExecutor(max_workers=4) as executor:
+            outstanding_futures = set()
+
+            for i in range(num_tests):
+                named_optimizer_config = optimizer_named_configs[i % num_optimizer_configs]
+                named_objective_function_config = objective_function_named_configs[i % num_objective_function_configs]
+
+                print("#####################################################################################################")
+                print(named_optimizer_config)
+                print(named_objective_function_config)
+
+                optimizer_evaluator_config = optimizer_evaluator_config_store.get_config_by_name(name="parallel_unit_tests_config")
+                optimizer_config = named_optimizer_config.config_point
+                objective_function_config = named_objective_function_config.config_point
+
+                optimizer_evaluator = OptimizerEvaluator(
+                    optimizer_evaluator_config=optimizer_evaluator_config,
+                    objective_function_config=objective_function_config,
+                    optimizer_config=optimizer_config
+                )
+
+                future = executor.submit(optimizer_evaluator.evaluate_optimizer)
+                outstanding_futures.add(future)
+
+            done_futures, outstanding_futures = concurrent.futures.wait(outstanding_futures, return_when=concurrent.futures.ALL_COMPLETED)

```

Comment:
you could also use ``joblib.Parallel`` for a slightly simpler interface that allows multiple backends.

Reply:
Thanks! What are some pros and cons of joblib vs. concurrent.futures aside from a simpler API?

Reply:
@amueller, I found this when researching comparisons: https://github.com/openvax/mhcflurry/issues/59

What are your thoughts?

Reply:
joblib indeed does batching of small jobs and probably does faster serialization of large arrays.
That comparison is from 2016 so I'm not sure it holds any more. concurrent.futures has a much more powerful interface in terms of lazy evaluation, joblib is basically a simple parallel for.